### PR TITLE
New autoprior

### PR DESCRIPTION
**Pull request by prappleizer**
Autoprior now is marginally less clunky as it uses the photutils.morphology package to generate guesses for things. 

Main changes: 

flux guess goes from jnp.sum(image) ---> cat.segment_flux 
(I tried both seg flux and kron flux and the former seemed closer to the best-fit answers for the three examples)

xc and yc guesses go from image center ---> x centroid and ycentroid (this one should be super solid now) 

theta now gets a guess --- I set the loc from 0--> theta guess (in radians) from photutils. Concentrations still 0. Should work? I assume with concentration still 0, it pretty much evenly covers the range? (so actually maybe nothing changed. We might need to add a bit of concentration to actually add importance to the loc. 

r_eff goes from image_dim/10 --> (cat.semimajoraxis_sigma/2), again, roughly dialed in using the example galaxies. 

All other params have full-range priors.

One additional change is that the photoutils thing can't read a masked image directly but can read a mask with the image, so I added optional arguments to the generate_x_prior functions to allow a mask to be added on the fly if desired; this is used by auto prior now.

### CODE REVIEW

File: pysersic/priors.py [hunk 1]
```diff
@@ -507,27 +514,33 @@ def generate_sersic_prior(image: jax.numpy.array,
...
+        r_eff_guess = (cat.semimajor_sigma/2).value
     
     r_loc = r_eff_guess
     r_scale = jnp.sqrt(r_eff_guess) 
     prior.set_truncated_gaussian_prior('r_eff', r_loc,r_scale, low = 0.5)
 
     prior.set_uniform_prior('ellip', 0,0.9)
-    prior.set_custom_prior('theta', dist.VonMises(loc = 0,concentration=0), reparam= infer.reparam.CircularReparam() )
+    theta_guess = cat.orientation.to(u.rad).value
+    prior.set_custom_prior('theta', dist.VonMises(loc = theta_guess,concentration=0), reparam= infer.reparam.CircularReparam() )
```
Comment (tbmiller-astro) [inline pysersic/priors.py:538]: I concentration of 0 implies flat distribution I think, see : https://en.wikipedia.org/wiki/Von_Mises_distribution, maybe set to 2 or something?

**Comment by tbmiller-astro** [comment]
Looks good to me! I think we might have to change it a little bit for MultiFitter, because it uses `autoprior` currently. So maybe just better to set those automatically?
